doc/headset.md

```diff
@@ -0,0 +1,430 @@
+# Genesys Cloud WebRTC SDK Headset Utility
+
+## SDK Headset Index
+* [Purpose of SdkHeadset]
+* [SdkHeadset]
+    * [Example usage]
+    * [Methods]
+    * [Events]
+
+
+## Purpose of SdkHeadset
+The SdkHeadset namespace connects directly to the new headset library that allows device call controls from at this moment three headset vendors. Plantronics/Poly, Sennheiser/EPOS and Jabra. It makes it possible for events from the headset to be properly handled and reflected in consuming apps. It also allows events from consuming apps to be properly reflected in the headset itself. This ensures the states of both the devices and consuming apps are always in sync
+
+--------
+
+## SdkHeadset
+
+### Example Usage
+
+``` ts
+
+/* async function for demonstration */
+async function testAnswerCall () {
+    /* This function handles a fromHeadset event. This will not cause the headset library to be called */
+    sdk.acceptPendingSession('123456789', 'softphone', true);
+    /*  async acceptPendingSession (params: IPendingSessionActionParams):       Promise<void> {
+        !params.fromHeadset && this.headset.answerIncomingCall(params.      conversationId);
+        await this.sessionManager.proceedWithSession(params);
+    }
+  */
+}
+
+async function testMuteCall () {
+    /* This function handles a from app event. This will cause the headset library to be called */
+    sdk.setAudioMute ('123456789', 'softphone', false | undefined);
+    /*  async acceptPendingSession (params: IPendingSessionActionParams):       Promise<void> {
+        !params.fromHeadset && this.headset.answerIncomingCall(params.      conversationId);
+        await this.sessionManager.proceedWithSession(params);
+    }
+
+    This in turn will call to the headset library which will update the headset's state to ensure everything is in sync
+
+    setMute (isMuted: boolean): Promise<void> {
+        return this.headsetLibrary.setMute(isMuted);
+    }
+  */
+}
+
+/* Examples of listening for events from the headset */
+sdk.headset.headsetEvents.subscribe(value: {
+    event: EmittedHeadsetEvents,
+    payload: {
+        name: string,
+        event: {`object containing various items from the headset`},
+        isMuted: boolean,
+        holdRequested: boolean,
+        isConnected: boolean,
+        isConnecting: boolean,
+        callback: Function,
+        toggle?: boolean,
+        code?: string
+    }
+} => {
+    switch (value.event) {
+        case 'deviceAnsweredCall':
+            this.trigger(value.event) | sdk.acceptPendingSession();
```

Comment:
So I copied this switch statement straight from web-dir and over there it will trigger the event name (like `deviceAnsweredCall` for example) and then that is listened for elsewhere.  Out of that context though, that definitely isn't clear... I'll remove the triggers and maybe just leave the `sdk.acceptPendingSession();` part. And I did mean to leave it in there to kind of show that a user could use any function they need to meet their needs after headset events